src/app/pages/progress-report/components/progress-report-chart/progress-report-chart.component.ts

```diff
@@ -0,0 +1,213 @@
+import {Component, Input, OnInit} from '@angular/core'
+import * as moment from 'moment'
+import {ChartOptions} from "../../models/chart";
+import {lineColors, markerColors, markerStroke} from "./chart-styles";
+import {getMockCurrentTime} from "./mock-tasks";
+
+@Component({
+  selector: 'progress-report-chart',
+  templateUrl: 'progress-report-chart.component.html',
+})
+export class ProgressReportChartComponent implements OnInit {
+  @Input()
+  tasksData: any
+
+  loading = true
+  chartData: any
+
+  referenceDate: number
+
+  public chartOptions: Partial<ChartOptions>;
+
+  constructor() {}
+
+  ngOnInit() {
+    const {tasks, referenceDate} = this.tasksData
+    this.referenceDate = referenceDate
+    this.chartData = this.convertTasksToChartData(tasks)
+    console.log(this.referenceDate)
+    console.log(this.chartData)
+    this.initCharts()
+  }
+
+  initCharts(){
+    this.chartOptions = {
+      series: this.chartData,
+      chart: {
+        id: "chart",
+        type: "area",
+        height: 230,
+        width: this.calculateWidth(),
+        toolbar: {
+          show: false,
+        },
+        foreColor: '#000',
+        fontFamily: 'Montserrat, sans-serif',
+      },
+      colors: lineColors,
+      stroke: {
+        curve: "straight"
+      },
+      dataLabels: {
+        enabled: false
+      },
+      fill: {
+        opacity: 1
+      },
+      markers: {...markerStroke, strokeOpacity: 1},
+      xaxis: {
+        type: 'category',
+        labels: {
+          show: true,
+          rotate: -45,
+          formatter: (val) => {
+            return this.getWeekNumberOfProtocol(val)
+          }
+        },
+        title: {
+          text: "Week",
+        },
+      },
+      yaxis: {
+        min: 0,
+        max: 100,
+        tickAmount: 2,
+        labels: {
+          show: true,
+          rotate: 0,
+          minWidth: 40,
+          formatter: function(val) {
+            if(val>50){return 'Full'}
+            if(val>1){return 'Half'}
+            return '';
+          },
+        },
+        title: {
+          text: "Task Completion",
+        },
+      },
+      tooltip: {
+        enabled: false,
+      },
+    }
+  }
+
+  calculateWidth(){
+    if(!this.chartData[0]) return "100%"
+    const numberOfItems = this.chartData[0].data.length // Math.floor(range/intervals["week"])
+    if(numberOfItems<13){
```

Comment:
Can we move this value `13` to a variable in case we want to make changes and for readability? And possibly the `30` below as well?